### PR TITLE
SONARJNKNS-250 Properly save SQ Scanner selection on jobs

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 1]
```diff
@@ -97,8 +94,16 @@
   /**
    * Identifies {@link SonarRunnerInstallation} to be used.
    * @since 2.0
+   * @deprecated since 2.5
+   */
+  @Deprecated
+  private transient String sonarRunnerName;
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: MINOR Remove the "transient" modifier from this field. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2065)
INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 2]
```diff
@@ -404,6 +409,14 @@ public Action getProjectAction(AbstractProject<?, ?> project) {
     return new SonarMarkerAction();
   }
 
+  protected Object readResolve() {
+    // Migrate old field to new field
+    if (sonarRunnerName != null) {
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: MINOR Remove this use of "sonarRunnerName"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 3]
```diff
@@ -404,6 +409,14 @@ public Action getProjectAction(AbstractProject<?, ?> project) {
...
+      sonarScannerName = sonarRunnerName;
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: MINOR Remove this use of "sonarRunnerName"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* MINOR 5 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [SonarRunnerBuilder.java#L204](https://github.com/SonarSource/jenkins-sonar-plugin/blob/a81cee90906c1fe389778136353a840efcfd035f/src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java#L204): Remove this use of "sonarRunnerName"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [SonarRunnerBuilder.java#L204](https://github.com/SonarSource/jenkins-sonar-plugin/blob/a81cee90906c1fe389778136353a840efcfd035f/src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java#L204): Remove this use of "sonarRunnerName"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MINOR 3 minor
* INFO 1 info

Watch the comments in this conversation to review them.

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java (hunk 1 above)
Comment (dbmeneses) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: Do transient vars get deserialized?

**Comment by dbmeneses** [comment]
one doubt, otherwise LGTM

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java (hunk 1 above)
Comment (henryju) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: Yep, they are simply not written back. I made a test.

**Pull request merged**